cspatients/util.py

```diff
@@ -87,42 +79,49 @@ def save_model_changes(data):
+        patiententry = PatientEntry.objects.get(
+            patient_id=changes_dict['patient_id']
+            )
+        del changes_dict['patient_id']
+        for label in changes_dict:
+            if label in PATIENTENTRY_FIELDS:
+                patiententry.__dict__[label] = changes_dict[label]
+            if label in PATIENT_FIELDS:
+                patiententry.__dict__[label] = changes_dict[label]
+        return 200
+    except PatientEntry.DoesNotExist:
+        return 400
+    except Patient.DoesNotExist:
+        return 400
+
+
+def save_model(post_data):
+    patient = PatientSerializer(
+        data=get_rp_dict(post_data, context="patient")
         )
-        if all_values.contains("name"):
-            current_patiententry.patient_id.name = all_values["name"]
-        if all_values.contains("age"):
-            current_patiententry.patient_id.age = all_values["age"]
-        if all_values.contains("operation"):
-            current_patiententry.operation = all_values["operation"]
-        if all_values.contains("gravpar"):
-            current_patiententry.gravpar = all_values["gravpar"]
-        if all_values.contains("comorbid"):
-            current_patiententry.comorbid = all_values["comorbid"]
-        if all_values.contains("indication"):
-            current_patiententry.indication = all_values["indication"]
-        if all_values.contains("decision_time"):
-            current_patiententry.decision_time = all_values["decision_time"]
-        if all_values.contains("discharge_time"):
-            current_patiententry.discharge_time = all_values["discharge_time"]
-        if all_values.contains("delivery_time"):
-            current_patiententry.delivery_time = all_values["delivery_time"]
-        if all_values.contains("urgency"):
-            current_patiententry.urgency = all_values["urgency"]
-        if all_values.contains("location"):
-            current_patiententry.location = all_values["location"]
-        if all_values.contains("outstanding_data"):
-            current_patiententry.outstanding_data = all_values["outstanding_data"]
-        if all_values.contains("clinician"):
-            current_patiententry.clinician = all_values["clinician"]
-        if all_values.contains("apgar_1"):
-            current_patiententry.apgar_1 = all_values["apgar_1"]
-        if all_values.contains("apgar_5"):
-            current_patiententry.apgar_5 = all_values["apgar_5"]
+    patiententry = PatientEntrySerializer(
+        data=get_rp_dict(post_data, context="patiententry")
+    )
+    if patient.is_valid():
+        patient.save()
+        if patiententry.is_valid():
+            patiententry.save()
+        return 201
+    else:
+        return 400
+
+
+def patient_has_delivered(post_data):
 
+    try:
+        patiententry = PatientEntry.objects.get(
+            patient_id=get_rp_dict(post_data)['patient_id']
+        )
+        patiententry.delivery_time = datetime.now()
```

Comment:
You need to call `patiententry.save()` after changing this value, otherwise django will not write the change to the database.